core/tls.cabal

```diff
@@ -160,6 +160,7 @@ Benchmark bench-tls
                    , QuickCheck >= 2
                    , tasty-quickcheck
                    , tls
+  ghc-options:       -Wall -fno-warn-unused-imports -fno-warn-orphans
```

Comment:
We don't need to be very meticulous in test/bench code.
Here I just finished the missing signatures because that was almost complete already.
Concerning orphan instances, I don't find @kazu-yamamoto's idea too bad because instances are located in two files only.